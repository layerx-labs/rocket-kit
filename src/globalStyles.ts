import { createGlobalStyle } from 'styled-components/macro';
import { rem} from "polished";
import { typography, colors } from './ions/variables';
import { device } from './ions/breakpoints';

const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
    scroll-behavior: smooth;
  }

  body {
    margin: 0;
    font-family: 'Space Grotesk', Verdana, Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: ${typography.regular};
    color: ${colors.black};
    font-feature-settings: 'ss04' on;
    -webkit-font-smoothing: antialiased;
  }

  h1,
  h1 span,
  h2,
  h2 span {
    margin: 0;
    font-weight: ${typography.regular};
  }

  h1,
  h1 span {
    font-family: 'Laforgestencil', Verdana, Arial, Helvetica, sans-serif;
    font-size: 3.5rem;
    line-height: 1.15;
  }

  h2,
  h2 span {
    font-size: 2rem;
    line-height: 1.2;
  }

  h3,
  h4 {
    margin: 0;
    font-size: 1rem;
    font-weight: 400;
  }

  span {
    font-family: ${typography.defaultFont};
    font-size: ${rem(typography.defaultSize)};
    font-feature-settings: 'ss04' on;
  }

  p {
    margin: 0;
    font-size: 1rem;
    line-height: 1.3;

    &:not(:last-child) {
      margin-bottom: ${rem('10px')};
    }
  }

  pre {
    border: 1px solid ${colors.grey500};
    border-radius: 4px;
    background-color: ${colors.grey300};
    padding: ${rem('15px')};
  }

  @media ${device.l} {
    h1,
    h1 span {
      font-size: 6rem;
      line-height: 1.1;
    }

    h2,
    h2 span {
      font-size: 2.5rem;
    }
  }
`;

export default GlobalStyle;
