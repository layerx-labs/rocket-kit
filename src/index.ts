// Test
export { Thing } from './thing';

// Ions
export { default as TaikaiLogo } from './ions/taikai-logo';

// Atoms
export { default as Avatar } from './atoms/avatar';
