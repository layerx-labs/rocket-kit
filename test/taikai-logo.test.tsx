import React from 'react';
import { render } from '@testing-library/react';
import { TaikaiLogo } from '../src/index';

describe('Taikai Logo', () => {
  it('renders', () => {
    const { asFragment } = render(<TaikaiLogo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
