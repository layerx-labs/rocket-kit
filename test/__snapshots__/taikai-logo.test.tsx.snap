// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taikai Logo renders 1`] = `
<DocumentFragment>
  <svg
    viewBox="0 0 500 86.48"
  >
    <defs>
      <lineargradient
        gradientUnits="objectBoundingBox"
        id="linear-gradient"
        x1="0.146"
        x2="0.854"
        y1="0.854"
        y2="0.146"
      >
        <stop
          offset="0"
          stop-color="#ef5867"
        />
        <stop
          offset="1"
          stop-color="#5031a8"
        />
      </lineargradient>
    </defs>
    <path
      d="M39.4 11.015V80.8a5.4 5.4 0 01-1.6 3.98 5.242 5.242 0 01-3.848 1.6A5.442 5.442 0 0128.4 80.8V11.015H5.573a5.464 5.464 0 01-3.914-1.6A5.138 5.138 0 010 5.575 5.288 5.288 0 011.659 1.6 5.464 5.464 0 015.573 0h56.786a5.464 5.464 0 013.914 1.6 5.288 5.288 0 011.658 3.98 5.138 5.138 0 01-1.658 3.848 5.464 5.464 0 01-3.914 1.6zm126.93 72.574a5.67 5.67 0 008.824 1.2 5.138 5.138 0 001.658-3.848 7.138 7.138 0 00-.4-2.256q-.266-.266-38.476-75.758A4.6 4.6 0 00133.427 0a4.692 4.692 0 00-4.644 3.052L90.441 78.681a4.976 4.976 0 00-.53 2.256 5.138 5.138 0 001.658 3.848 5.466 5.466 0 003.914 1.6h36.486a5.242 5.242 0 003.848-1.6 5.5 5.5 0 00-3.84-9.42H104.5l28.922-56.92zM223.121 0a5.464 5.464 0 00-3.92 1.6 5.288 5.288 0 00-1.658 3.98V80.8a5.288 5.288 0 001.658 3.98 5.6 5.6 0 007.828 0 5.288 5.288 0 001.658-3.98V5.575a5.288 5.288 0 00-1.652-3.975 5.464 5.464 0 00-3.914-1.6zm109.5.016a5.29 5.29 0 00-4.068 1.432l-47.6 42.826V5.667a5.292 5.292 0 00-1.658-3.98 5.6 5.6 0 00-7.828 0 5.292 5.292 0 00-1.658 3.98V80.9a5.29 5.29 0 001.658 3.98 5.6 5.6 0 007.828 0 5.29 5.29 0 001.658-3.98V59.3l11.326-10.2 36.04 35.776a1.188 1.188 0 00.8.664 5.614 5.614 0 003.582.93 5.236 5.236 0 003.448-1.6 5.03 5.03 0 001.66-3.716 5.252 5.252 0 00-1.4-3.848l-.4-.4-35.4-35.28L336 9.731a5.29 5.29 0 001.848-3.9 5.462 5.462 0 00-1.428-3.974 5.466 5.466 0 00-3.8-1.842zM494.429 0a5.464 5.464 0 00-3.914 1.6 5.288 5.288 0 00-1.658 3.98V80.8a5.288 5.288 0 001.658 3.98 5.6 5.6 0 007.828 0A5.288 5.288 0 00500 80.8V5.575a5.288 5.288 0 00-1.657-3.975 5.462 5.462 0 00-3.914-1.6zm-56.8 83.586a5.67 5.67 0 008.822 1.2 5.138 5.138 0 001.658-3.848 7.148 7.148 0 00-.4-2.256q-.266-.266-38.476-75.758A4.6 4.6 0 00404.721 0a4.692 4.692 0 00-4.644 3.052l-38.344 75.629a4.976 4.976 0 00-.53 2.256 5.138 5.138 0 001.658 3.848 5.464 5.464 0 003.914 1.6h36.486a5.244 5.244 0 003.848-1.6 5.5 5.5 0 00-3.84-9.42H375.8l28.92-56.92z"
      data-name="Path 173"
      fill="url(#linear-gradient)"
      id="Path_173"
      transform="translate(-.001 -.003)"
    />
  </svg>
</DocumentFragment>
`;
